src/App.jsx

```diff
@@ -21,25 +25,81 @@ function App() {
         wordDisplay.push("_");
       }
     }
+
     return wordDisplay.toString();
   };
 
+  //5. Add logic and state to track whether the user has guessed all letters of the word and how many guesses the user has left (can start with 10).
+  const [gameRounds, setGameRounds] = useState(10);
+  // const [score, setScore] = useState(0);
+
+  //6. If the user guesses all letters correctly, tell them they have won. If the user runs out of guesses, reveal the word and tell them they have lost.
   // create additional function to power the
 
+  const checkWinCondition = () => {
+    //create s shallow copy of the guessed letters
+    const currGuessedLetters = [...guessedLetters];
+
+    for (let letter of currWord) {
+      if (!currGuessedLetters.includes(letter)) {
+        return false;
+      }
+    }
+    return true;
+    // returns true if guessedletters contains all letters in current word.
+  };
+  const checkWin = checkWinCondition();
```

Comment:
```suggestion
  const hasPlayerWon = checkWinCondition();
```
That would be a bit better in naming I think. I think you could consider defining the variable in the `endGame` function, as it isn't used anywhere else